deltachat-ios/Chat/AudioController.swift

```diff
@@ -85,6 +85,34 @@ open class AudioController: NSObject, AVAudioPlayerDelegate, AudioMessageCellDel
             cell.audioPlayerView.setDuration(duration: player.currentTime)
         }
     }
+    
+    public func getAudioDuration(messageId: Int, successHandler: @escaping (Int, Double) -> Void) {
+        let message = DcMsg(id: messageId)
+        if playingMessage?.id == messageId {
+            // irgnore messages that are currently playing or recently paused
+            return
+        }
+        
+        if let fileURL = message.fileURL {
```

Comment:
before trying to get the duration from the file (which may be expensive), we should check [`dc_msg_get_duration()`](https://c.delta.chat/classdc__msg__t.html#aaedb426314f3d701d606ba3daa24fc1a) that _may_ also return the duration. 

only if that return 0 for "unknown", we should parse the file.

both checks, however, should be async inside `getAudioDuration()`

Reply:
nice, I wasn't aware of that api